info_encoding.md

```diff
@@ -119,25 +111,89 @@ function encode(l List) Buffer {
 }
 ```
 
-and in Go (untested):
+### Go
 
-```
+```go
 import "encoding/binary"
 
-func Encode(l [][]byte) []byte {
+func binEncodeUint16(b []byte, i uint16) {
+	binary.LittleEndian.PutUint16(b, i)
+}
+
+func Encode(list [][]byte) []byte {
 	var (
 		out []byte
 		l  = make([]byte, 2)
 	)
-	
-	for _, elem := range l {
-		binary.BigEndian.PutUint16(l, uint16(len(elem)))
-		append(out, l...)
-		append(out, elem...)
+
+	for _, elem := range list {
+		binEncodeUint16(l, uint16(len(elem)))
+		out = append(out, l...)
+		out = append(out, elem...)
 	}
 
 	return out
 }
 ```
 
-[Code]: #code
+### JavaScript
+
+```js
+function binEncodeUInt16(target, number) {
+    target.writeUInt16BE(number, 0)
+}
+
+function encode (list) {
+  var out = Buffer.alloc(0)
+
+  for (let buf in list) {
+    const length = Buffer.alloc(2)
+	binEncodeUInt16(length, buf.length)
+
+    out = Buffer.concat([out, length, buf])
+  }
+
+  return out
+}
+```
+where
+- `list` is an Array of Buffers
+- `encode` returns a Buffer
+
+## Examples
+
+To derive the read key from the message key, we perform a derivation with the info argument of HKDF.Expand set to the encoding of the following key-value list, displayed here to look a bit like a JSON object:
+```js
+{
+	"purpose": "box2",
+	"type": "read key",
+	"feed": "@feedID", // this is a placeholder
+	"prev": "%msgID"   // this is a placeholder
+}
+```
+This would be list-encoded by alternating between keys and values:
+```
+("purpose", "box2", "type", "read key", "feed", "@feedID", "prev", "%msgID")
```

Comment:
I think you need to say explicitly:

> here the "schema" is the expected keys and their values, along with **the order in which they're encoded**

Reply:
You haven't talked here about how these strings were converted to buffers. Should I know how to do this?